src/fds.rs

```diff
@@ -222,12 +222,16 @@ pub fn set_cloexec(fd: RawFd, should_set: bool /* = true */) -> c_int {
 
 /// Wide character version of open() that also sets the close-on-exec flag (atomically when
 /// possible).
-pub fn wopen_cloexec(pathname: &wstr, flags: i32, mode: libc::c_int) -> RawFd {
+pub fn wopen_cloexec(
+    pathname: &wstr,
+    flags: OFlag,
+    mode: nix::sys::stat::Mode,
+) -> nix::Result<OwnedFd> {
     open_cloexec(wcs2zstring(pathname).as_c_str(), flags, mode)
 }
 
```

Comment:
in general it's better to use checked versions and `unwrap()` to panic on failure.

Reply:
What are you referring to?

Reply:
meh github doesn't really support fine grained code review outside the web UI. I'd prefer to use email.

I was talking about

```diff

 /// Narrow versions of wopen_cloexec.
-pub fn open_cloexec(path: &CStr, flags: i32, mode: libc::c_int) -> RawFd {
+pub fn open_cloexec(path: &CStr, flags: i32, mode: libc::c_int) -> nix::Result<RawFd> {
+    let flags = unsafe { OFlag::from_bits_unchecked(flags) };
+    let mode = unsafe { nix::sys::stat::Mode::from_bits_unchecked(mode as u32) };
```

which are gone with later changes